src/search.rs

```diff
@@ -214,6 +214,49 @@ struct GoalFrame {
     cut_level: usize,
 }
 
+/// A single assignment of variables to terms.
+///
+/// The variables returned in this struture correspond to the goals via [ordinals](crate::ast::Var::ord).
+/// i.e. the assignment for a variable `x: Var` is stored at index
+/// `x.ord()`.
+///
+/// Names of variables are not stored here. They can be resolved using the matching Scope.
+///
+/// Note that even in valid solutions, not all variables need to have an assignment. For
+/// example, given the rule
+///
+/// ```prolog
+/// anything(X).
+/// ```
+///
+/// the query `anything(X)` will be true for any `X`, hence even a valid solution won't have an
+/// assignment for this variable.
+#[derive(PartialEq, Debug, Clone)]
+pub struct Solution(pub Vec<Option<ast::Term>>);
+
+impl Solution {
+    /// Returns a reference to the vector containing all assignments of terms to variables.
+    /// The vector's indices correspond to ordinals of the variables.
+    pub fn vars(&self) -> &Vec<Option<ast::Term>> {
+        &self.0
+    }
+
+    /// Iterates over all variables and their assigned terms.
+    pub fn iter_vars(&self) -> impl Iterator<Item = (Var, &Option<ast::Term>)> {
+        self.0
+            .iter()
+            .enumerate()
+            .map(|(i, term)| (Var::from_ord(i), term))
+    }
+
+    /// Returns the term matching this variable
+    pub fn get(&self, var: Var) -> Option<&ast::Term> {
+        self.iter_vars()
+            .find(|(v, _t)| v == &var)
+            .and_then(|(_v, term)| term.as_ref())
```

Comment:
This can just be the more efficient `self.0[v.ord()].as_ref()`.

Reply:
Ouch, don't sleep and code I guess.

I added some unit tests as an apology.